### PR TITLE
8343532: Test test/hotspot/jtreg/runtime/cds/appcds/jigsaw/addmods/AddmodsOption.java fails on Linux ppc64le after JDK-8319343

### PR DESCRIPTION
**Pull request by MBaesken**
Test test/hotspot/jtreg/runtime/cds/appcds/jigsaw/addmods/AddmodsOption.java fails on Linux ppc64le after [JDK-8319343](https://bugs.openjdk.org/browse/JDK-8319343) with error message
"Unrecognized VM option 'EagerJVMCI" .
The test needs jvmci but this is not supported/available on Linux ppc64le.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8343532](https://bugs.openjdk.org/browse/JDK-8343532): Test test/hotspot/jtreg/runtime/cds/appcds/jigsaw/addmods/AddmodsOption.java fails on Linux ppc64le after JDK-8319343 (**Bug** - P4)

### Reviewers
 * [Lutz Schmidt](https://openjdk.org/census#lucy) (@RealLucy - **Reviewer**) Review applies to [1d0d981d](https://git.openjdk.org/jdk/pull/21873/files/1d0d981d24cae3b2d7c2fec8ea5dc6bf529ec24d)
 * [Martin Doerr](https://openjdk.org/census#mdoerr) (@TheRealMDoerr - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21873/head:pull/21873` \
`$ git checkout pull/21873`

Update a local copy of the PR: \
`$ git checkout pull/21873` \
`$ git pull https://git.openjdk.org/jdk.git pull/21873/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21873`

View PR using the GUI difftool: \
`$ git pr show -t 21873`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21873.diff">https://git.openjdk.org/jdk/pull/21873.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21873#issuecomment-2454564589)
</details>

### CODE REVIEW

**Review by RealLucy — APPROVED** [review]
Looks good. And trivial.

**Comment by TheRealMDoerr** [comment]
Shouldn't we only disable the part which runs with -XX:+EagerJVMCI?

**Comment by MBaesken** [comment]
> Shouldn't we only disable the part which runs with -XX:+EagerJVMCI?

Maybe.  But as it is, the test AddmodsOption currently needs JVMCI so the require statement is correct.

**Comment by MBaesken** [comment]
> Well, disabling all tests is of course correct. But does that make sense? JVMCI seems to be only used here:
> 
> https://github.com/openjdk/jdk/blob/8d6cfba37fe641e35886fdba536f5b2f1709e87b/test/hotspot/jtreg/runtime/cds/appcds/jigsaw/addmods/AddmodsOption.java#L139

In WhiteBox there is isJVMCISupportedByGC, so should I use this for the relevant part?
`
WhiteBox.getWhiteBox().isJVMCISupportedByGC() `

**Review by TheRealMDoerr — CHANGES_REQUESTED** [review]
Please only disable the part which uses JVMCI.
I think it could be done like here: https://github.com/openjdk/jdk/blob/8d6cfba37fe641e35886fdba536f5b2f1709e87b/test/hotspot/jtreg/compiler/uncommontrap/Decompile.java#L62

**Comment by MBaesken** [comment]
> Please only disable the part which uses JVMCI.  I think it could be done like here:

Yes but quite a LOT of change for such a simple check.  And I have to change also from 'run driver' because otherwise the WhiteBox cannot be enabled.
And then the java version check does not work any more (but I guess this is probably done in other tests too?) ...

**Comment by MBaesken** [comment]
Maybe we should simply check for ppc ? The other platforms seem to have JVMCI. Then we can avoid all this bad overhead.

**Comment by TheRealMDoerr** [comment]
> Maybe we should simply check for ppc ? The other platforms seem to have JVMCI. Then we can avoid all this bad overhead.

Only x86_64, aarch64 and riscv64 have JVMCI support.

**Comment by TheRealMDoerr** [comment]
> And then the java version check does not work any more (but I guess this is probably done in other tests too?) ...

Thanks for implementing the check! What's the problem regarding the version check? Do you the purpose of it?

**Comment by TheRealMDoerr** [comment]
Alternatively, we could check the output for "Unrecognized VM option 'EagerJVMCI'".

File: test/hotspot/jtreg/runtime/cds/appcds/jigsaw/addmods/AddmodsOption.java [hunk 1]
```diff
@@ -107,7 +114,7 @@ public static void main(String[] args) throws Exception {
         oa.shouldHaveExitValue(0)
           .shouldContain("--add-modules module name(s) specified during runtime but not found in archive: jdk.jconsole")
           // version of the jdk.httpserver module, e.g. java 22-ea
-          .shouldMatch(versionPattern)
+          //.shouldMatch(versionPattern)
```
Comment (iklam) [inline test/hotspot/jtreg/runtime/cds/appcds/jigsaw/addmods/AddmodsOption.java:117]: Why are you taking this out?

**Comment by MBaesken** [comment]
> > And then the java version check does not work any more (but I guess this is probably done in other tests too?) ...
> 
> Thanks for implementing the check! What's the problem regarding the version check? Do you know the purpose of it?

It does not work any more after the changes.
Maybe the change from 'run driver' caused this.  I am not sure.

**Comment by MBaesken** [comment]
> > Maybe we should simply check for ppc ? The other platforms seem to have JVMCI. Then we can avoid all this bad overhead.
> 
> Only x86_64, aarch64 and riscv64 have JVMCI support.

So ppc64 and s390x do not have it. Would probably be much simpler to check for this.
Or adjust the try block where already some kind of check was present 

```
oa.shouldHaveExitValue(1)
                  .shouldContain("Cannot use JVMCI compiler: No JVMCI compiler found");
```

but this does not help in our case.

**Comment by TheRealMDoerr** [comment]
> So ppc64 and s390x do not have it.

Plus 32 bit platforms. Don't forget that it's possible to configure JVMCI out on the supported platforms.

**Comment by MBaesken** [comment]
Okay then let us keep the WhiteBox based check. I also brought back the version check and adjusted it to what we had already in 
test ModuleOption, this works also with my version number and some other version numbers that are possible
https://github.com/openjdk/jdk/blob/4fc6d4135e795d18a024a6035908f380b81082d1/test/hotspot/jtreg/runtime/cds/appcds/jigsaw/module/ModuleOption.java#L44

**Review by TheRealMDoerr — APPROVED** [review]
Thank you! LGTM.

**Comment by MBaesken** [comment]
Thanks for the reviews !
Maybe we should add something easier for checking availability of some 'JVM feature'  like jvmci,  then the whole WhiteBox stuff.
But that's for another change.

/integrate

**Pull request closed without merging**